hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/SimulatedUpdater.java

```diff
@@ -0,0 +1,239 @@
+import com.hedera.services.transaction.store.contracts.HederaWorldUpdater;
+import com.hedera.services.transaction.store.contracts.UpdateTrackingLedgerAccount;
+import com.hedera.services.transaction.store.contracts.WorldStateAccount;
+import com.hedera.services.transaction.store.contracts.WorldStateTokenAccount;
+
+@Data
+public class SimulatedUpdater implements HederaWorldUpdater {
+
+    private ContractRepository contractRepository;
+    private AliasesResolver aliasesResolver;
+    private SimulatedEntityAccess entityAccess;
+    private CodeCache codeCache;
+
+    private int numAllocatedContractIds = 0;
+    private long sbhRefund = 0L;
+    private long newContractNumId;
+
+    private final List<Address> provisionalContractCreations = new LinkedList<>();
+    protected final Set<Address> deletedAccounts = new HashSet<>();
+    protected final Map<Address, UpdateTrackingLedgerAccount<Account>> updatedAccounts = new HashMap<>();
+
+    //FUTURE WORK finish implementation when we introduce StackedUpdaters
+    public SimulatedUpdater updater() {
+        return new SimulatedUpdater();
+    }
+
+    @Override
+    public EvmAccount createAccount(final Address addressOrAlias, final long nonce, final Wei balance) {
+        final var address = aliasesResolver.resolveForEvm(addressOrAlias);
+        final var newMutable = new UpdateTrackingLedgerAccount<>(address);
+        newMutable.setNonce(nonce);
+        newMutable.setBalance(balance);
+
+        return new WrappedEvmAccount(track(newMutable));
+    }
+
+    @Override
+    public EvmAccount getAccount(Address address) {
+        final var extantMutable = updatedAccounts.get(address);
+        if (extantMutable != null) {
+            return new WrappedEvmAccount(extantMutable);
+        } else if (deletedAccounts.contains(address)) {
+            return null;
+        } else {
+            final var origin = get(address);
+            if (origin == null) {
+                return null;
+            }
+            final var newMutable = new UpdateTrackingLedgerAccount<>(origin);
+            return new WrappedEvmAccount(track(newMutable));
+        }
+    }
+
+    @Override
+    public void deleteAccount(Address addressOrAlias) {
+        final var address = aliasesResolver.resolveForEvm(addressOrAlias);
+        deletedAccounts.add(address);
+        updatedAccounts.remove(address);
+    }
+
+    @Override
+    public Collection<? extends Account> getTouchedAccounts() {
+        return new ArrayList<>(getUpdatedAccounts().values());
+    }
+
+    @Override
+    public Collection<Address> getDeletedAccountAddresses() {
+        return new ArrayList<>(getDeletedAccounts());
+    }
+
+    @Override
+    public void revert() {
+        getDeletedAccounts().clear();
+        getUpdatedAccounts().clear();
+
+        while (numAllocatedContractIds != 0) {
+            numAllocatedContractIds--;
+            newContractNumId--;
+        }
```

Comment:
Can't you just do subtraction to calculate?

```suggestion
        newContractNumId = newContractNumId - numAllocatedContractIds;
        numAllocatedContractIds = 0;
```